src/python/pants/jvm/package/deploy_jar.py

```diff
@@ -201,9 +209,14 @@ async def package_deploy_jar(
         Digest, AddPrefix(cat_and_repair.output_digest, str(output_filename.parent))
     )
 
+    reproducible_digest = renamed_output_digest
+
+    if field_set.reproducible:
+        reproducible_digest = await Get(Digest, StripJarRequest(renamed_output_digest))
```

Comment:
Will actually need to pass the filename it looks like. Will always be singular.

Reply:
Actually, the original idea was to be able to run multiple files through this because ClasspathRequest might have multiple files in one digest. Especially if the same rule is going to be used with https://github.com/pantsbuild/pants/issues/13879.

Reply:
fixed the missing `filenames`. parameter